### ISSUE TITLE
Azure Pipelines agent in DMZ cannot ping back Azure DevOps Server

### ISSUE THREAD

**Issue by ahmad2x4** [microsoft/azure-pipelines-agent#2277]
I am working with a company who are planning to upgrade to On-premises Azure DevOps 2019 Servers and start using Azure Pipeline. For the security reasons, they want to keep Azure DevOps inside their intranet. Production servers are in DMZ network and they want to be able to utilize Azure Pipeline to release to the production servers. 
According to documentation:

_Periodically, the agent checks to see if a new job request has been posted for it in the job queue in Azure Pipelines/TFS. When a job is available, the agent downloads the job as well as a job-specific OAuth token._ [Microsoft Docs](https://docs.microsoft.com/en-us/azure/devops/pipelines/agents/agents?view=azure-devops#communication)

By definition and again for the security reasons any network connection initiated from DMZ to the internal network will be blocked. Hence Azure Pipeline Agent won't be able to ping back home.

![image](https://user-images.githubusercontent.com/1284228/58444357-2f696280-813b-11e9-9e25-9e1f0c3f8f63.png)

Is there any solution to resolve this issue without Moving Azure DevOp server in DMZ?

**Comment by TingluoHuang** [comment]
@ahmad2x4 
Why the agent is in the DMZ but not your Azure DevOps Server? the Azure DevOps Server holds all your source code, work item and build artifacts, it should be treated as highly secure assets.

the configuration we saw most customer has is the Azure DevOps server is in DMZ, the agent is not in the DMZ.

the agent was design to better serve that configuration.

**Comment by TingluoHuang** [comment]
Closing for now, please reopen if need any help.

**Comment by Necriis** [comment]
@TingluoHuang I don't understand your answer. The DMZ is outside of the LAN because it's potentially unsafe, so the last place where I want to put my DevOps server, is in the DMZ, precisely because it should be treated as highly secure assets...
@ahmad2x4 I think the only possibility is to setup a specific rule in your firewall to allow your agent to ping back your DevOps Server.
